near-sdk/src/json_types/public_key.rs

```diff
@@ -152,29 +154,33 @@ impl std::str::FromStr for Base58PublicKey {
     }
 }
 
-#[non_exhaustive]
 #[derive(Debug)]
-pub enum ParsePublicKeyError {
+pub struct ParsePublicKeyError {
+    kind: ParsePublicKeyErrorKind,
+}
+
+#[derive(Debug)]
+enum ParsePublicKeyErrorKind {
     InvalidLength(usize),
     Base58(B58Error),
     UnknownCurve,
 }
 
 impl std::fmt::Display for ParsePublicKeyError {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            Self::InvalidLength(l) => {
+        match self.kind {
+            ParsePublicKeyErrorKind::InvalidLength(l) => {
                 write!(f, "Invalid length of the public key, expected 32 got {}", l)
             }
-            Self::Base58(e) => write!(f, "Base58 decoding error: {}", e),
-            Self::UnknownCurve => write!(f, "Unknown curve kind"),
+            ParsePublicKeyErrorKind::Base58(e) => write!(f, "Base58 decoding error: {}", e),
+            ParsePublicKeyErrorKind::UnknownCurve => write!(f, "Unknown curve kind"),
```

Comment:
Unrelated to the PR, but let's switch error messages to start with the lowercase letter, as recommended by the API guidelines and stdlib: https://doc.rust-lang.org/beta/std/error/trait.Error.html.